widgetssdk/src/main/java/com/glia/widgets/chat/adapter/ChatMessage.kt

```diff
@@ -0,0 +1,53 @@
+package com.glia.widgets.chat.adapter
+
+import androidx.annotation.VisibleForTesting
+import com.glia.widgets.chat.adapter.ChatMessage.Participant.Companion.toWidgetsType
+import com.glia.androidsdk.chat.ChatMessage as CoreChatMessage
+import org.json.JSONObject
+
+/**
+ * Represents a chat message in the Glia Widgets SDK.
+ *
+ * @property id The unique identifier of the message.
+ * @property content The content of the message.
+ * @property timestamp The timestamp of when the message was received by the server, in milliseconds.
+ * @property deliveredAt The date when the message was marked as read, or `null` if not marked yet.
+ * @property attachment The attachment associated with the message, or `null` if none exists.
+ * @property metadata Custom payloads for tailoring the visitor experience, or `null` if none exists.
+ * @property senderType The type of sender (visitor or operator).
+ */
+class ChatMessage(
+    val id: String,
+    val content: String,
+    val timestamp: Long,
+    val deliveredAt: String?,
+    val attachment: MessageAttachment?,
+    val metadata: JSONObject?,
+    val senderType: Participant
+) {
+    constructor(chatMessage: CoreChatMessage) : this(
```

Comment:
Can we make this constructor internal? Or integrators need it?

Reply:
Fixed ef470376e3f759d92b1eba9be9cfdf9fba107080